### PR TITLE
Add a web application for building detection from drone imagery using a trained U-Net model.

### PR DESCRIPTION
**Pull request by yiihuang**
### 🚀 Feature: Building Detection Interface with U-Net Integration

This PR introduces a complete frontend interface for real-time building detection powered by a trained U-Net model. Key features include:

* 🖼️ **Image Upload Interface**: Supports drag-and-drop and click-to-upload functionality for ease of use.
* 🤖 **AI-Based Processing**: Utilizes a trained U-Net model to perform real-time building segmentation.
* 📊 **Visualization Tools**: Displays the original image, prediction mask, and an overlay to help users interpret results effectively.
* 💾 **Download Capability**: Enables users to download processed images for offline use or further analysis.
* 📱 **Responsive Design**: Fully optimized for both desktop and mobile devices to ensure accessibility across platforms.

This update significantly enhances user experience and model interpretability for building detection workflows.

### CODE REVIEW

**Comment by BinaryNavigator07** [comment]
fix the dir structure

**Comment by BinaryNavigator07** [comment]
Move the streamlit app in a separate folder in such a way that the streamlit app can be used as a quick demo, and we still have a clean codebase for the cloud and flask or fastapi deployment

File: data_setup.py [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+    )
+    
+    val_loader = DataLoader(
+        val_dataset, 
+        batch_size=batch_size, 
+        shuffle=False, 
+        num_workers=num_workers
+    )
+    
+    return train_loader, val_loader
```
[the remark this answers is not in the document]
Comment (yiihuang) [reply data_setup.py]: Fixed.

File: example_2.tif
[the remark this answers is not in the document]
Comment (yiihuang) [reply example_2.tif:1]: Fixed.

File: example_3.tif
[the remark this answers is not in the document]
Comment (yiihuang) [reply example_3.tif:1]: Fixed.
